config/rbac/role.yaml

```diff
@@ -98,6 +106,14 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
+  - cluster.x-k8s.io
+  resources:
+  - machinedeployments
+  verbs:
+  - get
```

Comment:
don't we need machinedeployments/finalizers as below?

Reply:
also these are already declared here https://github.com/kubernetes-sigs/cluster-api/pull/5191/files#diff-6d4c4e29ec096149781536bb9118fde8242c1901df764e3eda3019ac8503d6c7R68?

Reply:
Yeah kind of. I have to add the kubebuilder tag (where depends on the outcome of the race condition discussion)

Reply:
I added it to the MD reconciler. (only need it for finalizer removal)